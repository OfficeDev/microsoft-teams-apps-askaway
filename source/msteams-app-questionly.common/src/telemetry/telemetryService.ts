import * as appInsights from "applicationinsights";
import { Contracts } from "applicationinsights";

let telemetryClient: appInsights.TelemetryClient;

export const initiateAndGetAppInsights = (
  applicationInsightsInstrumentationKey: string
) => {
  // Set up app insights
  appInsights
    .setup(applicationInsightsInstrumentationKey)
    .setAutoDependencyCorrelation(true)
    .setAutoCollectRequests(true)
    .setAutoCollectPerformance(true, true)
    .setAutoCollectExceptions(true)
    .setAutoCollectDependencies(true)
    .setAutoCollectConsole(true, true)
    .setUseDiskRetryCaching(true)
    .setSendLiveMetrics(true)
    .setDistributedTracingMode(appInsights.DistributedTracingModes.AI);
  appInsights.start();

  telemetryClient = appInsights.defaultClient;
  telemetryClient.addTelemetryProcessor(addUserContext);

  return telemetryClient;
};

function addUserContext(envelope: Contracts.Envelope): boolean {
  const data = envelope.data["baseData"];
  if (data?.properties?.userAadObjectId) {
    telemetryClient.context.tags["ai.user.id"] =
      data.properties.userAadObjectId;
  }
  return true;
}
