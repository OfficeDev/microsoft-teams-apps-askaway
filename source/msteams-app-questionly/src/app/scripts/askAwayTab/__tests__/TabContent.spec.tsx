/**
 * @jest-environment jsdom
 */

import * as React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { TabContent } from '../TabContent';
import { telemetryService } from '../../telemetryService';
import { HttpService } from '../shared/HttpService';
import Helper from '../shared/Helper';
import { i18next } from '../shared/i18next';
import TabCreateSession from '../TabContent/TabCreateSession';
import PostNewQuestions from '../TabContent/PostNewQuestions';
import NoQuestionDesign from '../TabContent/NoQuestionDesign';
import TabQuestions from '../TabContent/TabQuestions';

configure({ adapter: new Adapter() });

describe('TabContent Component', () => {
    const httpServiceIns = new HttpService(telemetryService.appInsights);
    const t = jest.fn();
    const tReady = true;

    beforeAll(() => {
        jest.mock('../shared/HttpService');
    });

    afterAll(() => {
        jest.resetAllMocks();
    });

    it('should render TabCreateSession when there is no active session', () => {
        const component = shallow(
            <TabContent t={t} tReady={tReady} i18n={i18next} teamsTabContext={{ entityId: '', locale: '' }} httpService={httpServiceIns} appInsights={telemetryService.appInsights} helper={Helper} />
        );

        expect(component.find(TabCreateSession)).toHaveLength(1);
    });

    it('should render PostNewQuestions when there is an active session', () => {
        const component = shallow(
            <TabContent t={t} tReady={tReady} i18n={i18next} teamsTabContext={{ entityId: '', locale: '' }} httpService={httpServiceIns} appInsights={telemetryService.appInsights} helper={Helper} />
        );
        component.setState({
            activeSessionData: {
                sessionId: 'some-id',
                title: '',
                isActive: false,
                dateTimeCreated: new Date(),
                hostUser: {
                    id: '',
                    name: '',
                },
                answeredQuestions: [],
                unansweredQuestions: [],
            },
        });

        expect(component.find(PostNewQuestions)).toHaveLength(1);
    });

    it('should render NoQuestionDesign when there is an active session and no questions', () => {
        const component = shallow(
            <TabContent t={t} tReady={tReady} i18n={i18next} teamsTabContext={{ entityId: '', locale: '' }} httpService={httpServiceIns} appInsights={telemetryService.appInsights} helper={Helper} />
        );
        component.setState({
            activeSessionData: {
                sessionId: 'some-id',
                title: '',
                isActive: false,
                dateTimeCreated: new Date(),
                hostUser: {
                    id: '',
                    name: '',
                },
                answeredQuestions: [],
                unansweredQuestions: [],
            },
        });

        expect(component.find(NoQuestionDesign)).toHaveLength(1);
    });

    it('should render TabQuestions when there is an active session and have questions', () => {
        const component = shallow(
            <TabContent t={t} tReady={tReady} i18n={i18next} teamsTabContext={{ entityId: '', locale: '' }} httpService={httpServiceIns} appInsights={telemetryService.appInsights} helper={Helper} />
        );
        component.setState({
            activeSessionData: {
                sessionId: 'some-id',
                title: '',
                isActive: false,
                dateTimeCreated: new Date(),
                hostUser: {
                    id: '',
                    name: '',
                },
                answeredQuestions: [{ id: '123' }],
                unansweredQuestions: [{ id: '456' }],
            },
        });

        expect(component.find(PostNewQuestions)).toHaveLength(1);
        expect(component.find(TabQuestions)).toHaveLength(1);
    });
});
