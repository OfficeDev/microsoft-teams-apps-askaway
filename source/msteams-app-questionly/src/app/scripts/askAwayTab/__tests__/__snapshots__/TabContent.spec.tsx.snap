// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabContent Component should match the snapshot 1`] = `
<div
  className="tab-content"
>
  <TabHeader
    activeSessionData={
      Object {
        "answeredQuestions": Array [],
        "dateTimeCreated": 2021-02-03T04:32:35.789Z,
        "hostUser": Object {
          "id": "",
          "name": "",
        },
        "isActive": false,
        "sessionId": "",
        "title": "",
        "unansweredQuestions": Array [],
      }
    }
    endSession={[Function]}
    refreshSession={[Function]}
    showTaskModule={[Function]}
    t={[MockFunction]}
  />
  <TabCreateSession
    showTaskModule={[Function]}
    t={[MockFunction]}
  />
</div>
`;
